### DISCUSSION TITLE
Flatpak support

### DISCUSSION THREAD

**Discussion by felipehw** [smplayer-dev/smplayer#39] [General]
[Flatpak](https://flatpak.org/) in being wildly adopted by the main Linux desktop project (GNOME), KDE, and several distros are enabling it (some depends almost exclusively on it like [Silverblue](https://silverblue.fedoraproject.org/)).

The main store of Flatpak is [Flathub](https://flathub.org/home). Many people search by new software there (directly at the site or via "app store"). So, if someone has his software in Flatpak + Flathub:

- Easier to new users discover it. (By App Store or [Flathub](https://flathub.org/home)).
- Easier to install. Just a click in the App Store (like Gnome/Ubuntu Software), in the same way as their other distro packages (rpm, debs, etc.). So, people don't need to know the site of the developer or the existence of AppImages, etc.
- Automatically updates.
- Better integration with Desktop (we can see it in the software center, with version, license, button to remove, etc.)
- The users don't need to be at the latest Linux distribution release to have the most updated software version (as is the case of a lot of distros).
- As said by *autocascade* [here](https://old-forum.smplayer.info/viewtopic.php?f=4&t=11124), "Linux is so fragmented it would seem to be a good way to distribute apps".

**Comment by fastrizwaan** [comment]
https://github.com/fastrizwaan/smplayer_flatpak/blob/main/info.smplayer.SMplayer.yml

**Reply by felipehw** [reply]
What do you think about publishing this at Flathub?

**Reply by smplayer-dev** [reply]
I'm definitely going to look into it. Thank you very much.

**Comment by fastrizwaan** [comment]
https://github.com/fastrizwaan/smplayer_flatpak/blob/main/README.md#build-and-install
```
flatpak install -y --user org.kde.Platform/x86_64/5.15 org.kde.Sdk/x86_64/5.15 
flatpak-builder --force-clean --install --user build-dir/ info.smplayer.SMplayer.yml
```

**Reply by smplayer-dev** [reply]
Unfortunately it doesn't work:
```
$ flatpak install -y --user org.kde.Platform/x86_64/5.15 org.kde.Sdk/x86_64/5.15 
Looking for matches…
error: No remote refs found similar to ‘org.kde.Platform/x86_64/5.15’
```

**Reply by felipehw** [reply]
Maybe you have to install flathub repo first.

**Reply by smplayer-dev** [reply]
Yes, it seems this is necessary first: 
`flatpak --user remote-add --if-not-exists flathub https://dl.flathub.org/repo/flathub.flatpakrepo `

**Reply by felipehw** [reply]
If Smplayer is offered through Flathub, this step would be done by the user.
Flathub is the "de facto" standard repo for Flatpak.

**Reply by madisonpeople** [reply]
Thanks, reading this thread helps me understand what been going on at System76 where I bought my 1st Linux designed hardware, running pop os, and so on. I am confident I can fix what did...I went ahead and loaded a problematic file, Error message-now reads
"...E: The repository 'http://ppa.launchpad.net/vikoadi/ppa/ubuntu impish Release' does not have a Release file.  ..."
I don't understand everything but I am learning.

**Comment by smplayer-dev** [comment]
I managed to build a flatpak file:
https://github.com/smplayer-dev/test-flatpak/releases/tag/v21.1.0.9609

**Reply by fastrizwaan** [reply]
great, could you please submit this at flathub. https://github.com/flathub/flathub/wiki/App-Submission

flathub needs appdata.xml please see this mpv appdata xml file or modify for smplayer. https://github.com/fastrizwaan/mpv_flatpak  Thank you.

**Comment by smplayer-dev** [comment]
I built another version: https://github.com/smplayer-dev/test-flatpak/releases/tag/v21.1.0.9616
It includes smtube and I fixed some problems with youtube-dl.
It would be great if people could test it, just in case I forgot something or there's something that doesn't work as expected.

**Reply by fastrizwaan** [reply]
Looks good:
1. video plays
2. subtitles downloads
3. smtube works - youtube videos play but this thing is a bit scary
![Screenshot from 2021-05-03 19-44-30](https://user-images.githubusercontent.com/1161682/116888226-d371b780-ac48-11eb-9a98-466f48b2e9f1.png)

**Reply by fastrizwaan** [reply]
Rather a less wordy prompt would be good:

YouTube Support needs youtube-dl addon. Install YouTube Support?
[  Yes ]  [  No  ]

**Reply by felipehw** [reply]
Works great here.

Maybe the submit process at Flathub gives you some good feedback to improve the package :D.
Finally, I'm using an updated SMplayer regardless my distro (Silverblue here).
🎉🎉🎉🎉🎉

**Comment by smplayer-dev** [comment]
It seems it doesn't work on Wayland.

**Comment by smplayer-dev** [comment]
More problems:
* Icons are not shown in the system tray.
* It seems dbus doesn't work.

**Reply by fastrizwaan** [reply]
https://github.com/fastrizwaan/Artha_flatpak/blob/main/net.sourceforge.Artha.yaml
```
finish-args:
  - --socket=wayland
  - --socket=fallback-x11
  - --device=dri
  # dbus session bus for artha
  - --own-name=org.artha.unique
```
https://github.com/flathub/org.qbittorrent.qBittorrent/blob/master/org.qbittorrent.qBittorrent.yaml
```
finish-args:
  - --socket=wayland
  - --socket=x11
  - --share=ipc
  - --share=network
  - --device=dri
  - --env=TMPDIR=/var/tmp # QtSingleApplication
  - --filesystem=host
  - --talk-name=org.kde.StatusNotifierWatcher
  - --talk-name=org.freedesktop.Notifications
  - --talk-name=org.freedesktop.PowerManagement
  - --talk-name=org.gnome.SessionManager
  - --own-name=org.kde.StatusNotifierItem-2-2
```

**Comment by topgunbg** [comment]
It doesn't work for me under Wayland - Fedora Silverblue 34 - it shows black video with audio.

The interesting thing is that when i enable 'Advanced - Show mpv in its own window' - the video is played no problem in separate mpv window

When choosing Video output driver - most don't work - with wlshm and drm - the video is played correctly but outside of the borders of the GUI of the player - like a separate undecorated window (not mpv).

I hope this helps!

**Comment by smplayer-dev** [comment]
[Version 21.1.0.9630](https://github.com/smplayer-dev/test-flatpak/releases/tag/v21.1.0.9630) now includes mplayer.

**Comment by smplayer-dev** [comment]
https://flathub.org/apps/details/info.smplayer.SMPlayer

**Comment by felipehw** [comment]
Thanks!
